src/server/main_service.cc

```diff
@@ -1056,6 +1056,20 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
               << " in dbid=" << dfly_cntx->conn_state.db_index;
   }
 
+  string_view cmd_name(cid->name());
+  bool is_write = (cid->opt_mask() & CO::WRITE) || cmd_name == "PUBLISH" || cmd_name == "EVAL" ||
+                  cmd_name == "EVALSHA";
+  if (cmd_name == "EXEC") {
+    // Check if one of the commands under exec is write command
+    for (auto& cmd : dfly_cntx->conn_state.exec_info.body) {
+      if (cmd.Cid()->opt_mask() & CO::WRITE) {
+        is_write = true;
+        break;
+      }
+    }
+  }
+  etl.AwaitPauseState(is_write);
```

Comment:
1. So if pause dispatch worked, this part would assume that all commands going through here are standalone? Why don't we just check instead of assuming?
2. Currently EXEC is always "write"

Reply:
3. It's better to use Service::VerifyCommandState because squashed commands don't use DispatchCommand

Reply:
1. by standalone do you mean not under script or multi transaction? Well no I do not assume this.. what I assume is that I am not in the middle of a transaction, i.e I did not execute few commands in multi and then client pause state was set and then I wait till pause is finished and will continue executing some other commands in the transaction.
2. why are you saying this? if exec has non wirte commands inside it is not write.. I just verified that with cli.
3. we do not enable squashed commands when pause is set, see my comment on what you wrote on collecting commands bellow.